docs/source/advanced_patterns.rst

```diff
@@ -0,0 +1,107 @@
+.. _advanced_communication_patterns:
+
+
+*******************************
+Advanced communication patterns
+*******************************
+
+Some advanced communication patterns are also implemented in osBrain. They are
+called channels, rather than sockets, as they are formed with multiple sockets.
+
+
+.. index:: async-req-rep
+.. _async_req_rep:
+
+Asynchronous Request-Reply channel
+==================================
+
+An asynchronous request-reply channel is very similar to a normal request-reply
+pattern, only in this case we want to process the reply asynchronously (i.e.:
+assign a handler for this reply and forget about it until it is received).
+
+This means that, when using an asynchronous request-reply channel, a handler
+must be specified not only for the replier but also for the requester.
+
+Example
+-------
+
+See the following example in which Bob sends an asynchronous request to Alice.
+Note that:
+
+- When Alice binds, she uses ``ASYNC_REP`` instead of ``REP``.
+- Bob is assigned a handler too to process Alice reply when it is received.
+- Bob sends the request and is automatically freed. It can log messages or do
+  other stuff.
+- When the reply is received, Bob automatically processes it.
+
+.. literalinclude:: ../../examples/async_req_rep.py
+
+That is the most basic example, but channels are a bit more flexible. When
+sending an asynchronous request we may want to manually specify a handler for
+the reply. This will overwrite the default handler specified when connecting to
+the ``ASYNC_REP`` server. Try to change the ``.send()`` call::
```

Comment:
"Try to x" sounds like the user won't be able to do so because some error will raise. Maybe change this to "Run the last example changing the .send() call", or simply "Change the .send() call".

Reply:
"Try" as in "experiment" or "do it yourself" (i.e.: our "prueba" not our "intenta"). Does it still sound weird to you?

Reply:
Meh. Just a bit.

I'd also say there is a very subtle difference between "try to change" and "try changing", and I'd rather go with the latter here. But there's also no need to argue too much about this, do whatever you want 😁.

Reply:
I like try changing too. :blush: